### ISSUE TITLE
[grafana]Grafana failing uprade to chart 6.44.0

### ISSUE THREAD

**Issue by singhsneetendra** [grafana/helm-charts#1986]
helm charts upgrade to version 6.44.0 show no errors, but no grafana pod instance shows up running.

**Comment by rdxmb** [comment]
I have the same problem with v6.44.2  with statefulset.

But as I render the `helm template` for myself, I see that no statefulset is deployed with this version.

Seems like the refactoring in https://github.com/grafana/helm-charts/commit/45f71da2c9055f6de23761434a8fc2472559ca53#diff-855fc83a8294ad9344d85e990013ad035e8a7fb6e359276ea20614c74f3220ae has an error.

**Comment by singhsneetendra** [comment]
@rdxmb that is true, I have replaced, persistence type from statefulset to pvc and that made upgrade to work. 

Changed 

 persistence:
-  type: statefulset
to 
 persistence:
-  type: pvc

**Comment by rdxmb** [comment]
The headless-service, however, uses the old `if`-condition

https://github.com/grafana/helm-charts/commit/45f71da2c9055f6de23761434a8fc2472559ca53#diff-773eb2f53d3326a4d503880ca8ab1c1accad1dfec4646c6b6bd94733390926d3R1

Could anybody explain me what's the magic with this `$sts` - Variable?

 @singhsneetendra Where did you find that is has to be `type: pvc` ?

**Comment by zanhsieh** [comment]
> The headless-service, however, uses the old `if`-condition
> 
> [45f71da#diff-773eb2f53d3326a4d503880ca8ab1c1accad1dfec4646c6b6bd94733390926d3R1](https://github.com/grafana/helm-charts/commit/45f71da2c9055f6de23761434a8fc2472559ca53#diff-773eb2f53d3326a4d503880ca8ab1c1accad1dfec4646c6b6bd94733390926d3R1)
> 
> Could anybody explain me what's the magic with this `$sts` - Variable?
> 
> @singhsneetendra Where did you find that is has to be `type: pvc` ?
> 
> EDIT: ok, it seems to be here:
> 
> https://github.com/grafana/helm-charts/blob/45f71da2c9055f6de23761434a8fc2472559ca53/charts/grafana/values.yaml#L297
> 
> I had not find it first because it was not in that commit above.

@singhsneetendra @rdxmb 
`statefulset`  is not a valid type in standard k8s manifest level. I have no idea who let it pass previous PR.

**Comment by sakvaka** [comment]
So, `StatefulSet` and `sts` are now the working values for `.Values.persistence.type` ? Is the plan to keep it like this (and update the documentation) or to restore the old behaviour?

**Comment by zanhsieh** [comment]
@sakvaka 
Already merged. See:

https://github.com/grafana/helm-charts/pull/2041/files

**Comment by rdxmb** [comment]
@zanhsieh I think @sakvaka's question is is about the default value in values.yaml , which is

```
persistence:
  type: pvc
```

https://github.com/grafana/helm-charts/pull/2041/files#diff-98c2472bc9183f58f15ae217e67d9bf3ee0e6e087dfb48ba82f6f073a4ba56b9L317

The same is in `README.md`
